### ISSUE TITLE
Editing ACLs

### ISSUE THREAD

**Issue by michalramus** [azlux/pymumble#123]
Could you implement ACLs editing and users mute/unmute features?

**Comment by azlux** [comment]
Hi,
mute/unmute should be already implemented : https://github.com/azlux/pymumble/blob/pymumble_py3/API.md#user-object
global ACL aren't implemented, it's not really the goal of this lib. I currently don't have time for that.

But I accept PR and can provide help.

**Comment by michalramus** [comment]
I need to edit ACLs from console application hosted on server. Could you tell me how can I do that in other way?

**Comment by azlux** [comment]
Only from the console ? I suppose it's for some automation ?
If you can script, I think mumo is the best : https://wiki.mumble.info/wiki/Mumo

**Comment by michalramus** [comment]
Sorry, I haven't explained myself clearly. I am going to create intercom system for video production. Example: user one is master, channel 1, 2, 3 - channel with groups e. g. camera operators/audio operators. As master user I have to talk to these channels, listen to them and change channels that I'm talking and listening in real time. I made small research and I think changing ACLs in real time is the best option.

I checked Mumo and it doesn't support managing ACLs

EDIT:  I'm going to create script in e. g. Python that reads commands from mqtt and makes changes in Mumble.

**Comment by azlux** [comment]
Ho, I think the "link" channel is what you're aiming. https://wiki.mumble.info/wiki/ACL_and_Groups/English#Link_Channel
I'm not sure to understand yours needs.

**Comment by michalramus** [comment]
Yes, and I need ACLs to manage links(sometimes I need to hear first channel, sometimes second, and sometimes speak to one of them).

**Comment by azlux** [comment]
Link is also a supported feature, it's not really a ACL : https://github.com/azlux/pymumble/blob/pymumble_py3/API.md#channel-object

**Comment by michalramus** [comment]
By default linked channels can listen and talk to each other. I need to disable listening first channel to second without disabling listening second channel to first and that's why I need ACLs.

**Comment by michalramus** [comment]
I'm going to make more complicated variation of this: https://www.rythechurchtechguy.com/mumble-effective-free-audio-communications-for-small-church-production/#amp_tf=Od%3A%20%251%24s&aoh=16652475842076&referrer=https%3A%2F%2F

**Comment by azlux** [comment]
Hi,
I'm not even sure if ACL is enough for your need.
I'm not currently working on pymumble, ACL is part of protobuff messages i've never worked on.
